File: WanderBee-Backend/chat-service/src/main/java/com/wanderbee/chatservice/model/ChatMessage.java
package com.wanderbee.chatservice.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.index.Indexed;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "chat_messages")
public class ChatMessage {

    @Id
    private String id;

    /** Text body or image URL */
    private String content;

    private String senderId;

    /** Null for group messages */
    private String recipientId;

    @Indexed
    private String roomId;

    @Builder.Default
    private LocalDateTime timestamp = LocalDateTime.now();

    @Builder.Default
    private MessageType type = MessageType.TEXT;
}

File: app/src/main/java/com/example/wanderbee/data/remote/models/chat/ChatMessage.kt
package com.example.wanderbee.data.remote.models.chat

/**
 * Matches the backend ChatMessage MongoDB document.
 * senderId / recipientId are the user's email (same value the gateway
 * extracts from the JWT and forwards as X-User-Id).
 */
data class ChatMessage(
    val id: String? = null,
    val content: String = "",
    val senderId: String = "",
    val recipientId: String? = null,
    val roomId: String = "",
    val timestamp: String? = null,   // ISO-8601 LocalDateTime from the backend
    val type: MessageType = MessageType.TEXT
)

enum class MessageType {
    TEXT, IMAGE, SYSTEM, DELETE
}